packages/manager/apps/pci-cold-archive/src/pages/containers/new/New.page.tsx

```diff
@@ -55,7 +56,38 @@ export default function ContainerNewPage() {
 
   const goBack = () => navigate('..');
 
-  const { createContainer, isPending } = useCreateContainer({
+  const {
+    mutateAsync: postSecretAsync,
+    isPending: isSecretPending,
+  } = usePostS3Secret({
+    projectId: project.project_id,
+    userId: form.selectedUser?.id,
+    userAccess: form.selectedUser?.s3Credentials?.access,
```

Comment:
By contract, `userId` & `userAccess` should always be defined

Reply:
yes and the api will return an error that will be displayed if some mandatory parameters are missing, that's the behavior we expect when we perform the api call

i don't think checking manually that userId & userAccess is defined then throw and display an error manually is better since it would mean that you don't trust the api behavior, it will also add more unecessary logic to the view

Reply:
But here, you are assigning non defined values that must be always be defined, by contrat (typing).
So, it come to do some plain JS

Reply:
the ts compiler is not reporting issues so i wouldn't call this "not compliant with ts engine"

Reply:
ts engine is not reporting issues because we don't use the `strict` mode, that should be enabled by default :)
But anyway, we can keep like this so.